packages/select-rich/src/LionSelectRich.js

```diff
@@ -200,31 +197,17 @@ export class LionSelectRich extends OverlayMixin(
     this.__setupInvokerNode();
     this.__setupListboxNode();
 
-    this._invokerNode.selectedElement = this.formElements[this.checkedIndex];
+    // Now that we have rendered our listbox, try setting the user defined modelValue again
+    if (this.__cachedUserSetModelValue) {
+      this.modelValue = this.__cachedUserSetModelValue;
+    }
 
+    this._invokerNode.selectedElement = this.formElements[this.checkedIndex];
     this.__toggleInvokerDisabled();
   }
 
   _requestUpdate(name, oldValue) {
     super._requestUpdate(name, oldValue);
-    if (
-      name === 'checkedValue' &&
-      !this.__isSyncingCheckedAndModelValue &&
-      this.modelValue &&
-      this.modelValue.length > 0
-    ) {
-      if (this.checkedIndex) {
-        // Necessary to sync the checkedIndex through the getter/setter explicitly
-        // eslint-disable-next-line no-self-assign
-        this.checkedIndex = this.checkedIndex;
-      }
-    }
-
-    if (name === 'modelValue') {
-      this.dispatchEvent(new CustomEvent('select-model-value-changed'));
```

Comment:
Untested, and just generally very nasty and bad practice. Better to not support this and force ourselves to clean up model-value-changed events to not be composed / bubble..